system/lib/libc/dynlink.c

```diff
@@ -580,16 +590,24 @@ void* dlopen(const char* file, int flags) {
 
 void emscripten_dlopen(const char* filename, int flags, void* user_data,
                        em_dlopen_callback onsuccess, em_arg_callback_func onerror) {
+  dbg("emscripten_dlopen: %s", filename);
   if (!filename) {
-    onsuccess(user_data, head);
+    onsuccess(user_data, head->dso);
```

Comment:
Is this a separate bugfix?

Reply:
Yes, I just noticed this as I was refactoring.